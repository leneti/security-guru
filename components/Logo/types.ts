import type { MouseEventHandler } from "react";

export interface LogoProps {
  iconOnly?: boolean;
  size?: number;
  vertical?: boolean;
  onClick?: MouseEventHandler<HTMLAnchorElement>;
  className?: string;
}
