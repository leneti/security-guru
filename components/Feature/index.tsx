export { default as Feature } from "./Feature";
export * from "./types";
