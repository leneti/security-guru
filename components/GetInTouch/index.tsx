export { default, type ContactForm } from "./GetInTouch";
