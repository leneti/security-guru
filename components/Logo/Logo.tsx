import Image from "next/image";
import Link from "next/link";
import { useGlobalStyles } from "@utils";
import Ikona_Oranzine from "@assets/logo/Ikona_Oranzine.webp";
import HLogotipas_Oranzine from "@assets/logo/HLogotipas_Oranzine.webp";
import Logotipas_Oranzine from "@assets/logo/Logotipas_Oranzine.webp";
import { LogoProps } from "./types";

const defaultSize = 50;

export default function Logo(props: LogoProps) {
  const { iconOnly, size, drawer, onClick } = props;
  const { classes: gClasses } = useGlobalStyles();

  const logoSize = size ?? defaultSize;

  if (iconOnly) {
    return (
      <Link href="/" onClick={onClick}>
        <Image
          alt="SG logo"
          src={Ikona_Oranzine}
          height={logoSize + 10}
          width={((logoSize + 10) / 2001) * 2000}
          unoptimized
        />
      </Link>
    );
  }

  if (drawer) {
    return (
      <Link href="/" onClick={onClick}>
        <Image
          alt="SG logo"
          src={HLogotipas_Oranzine}
          height={logoSize * 0.75}
          width={(logoSize / 341) * 2000 * 0.75}
          unoptimized
        />
      </Link>
    );
  }

  return (
    <>
      <Link
        href="/"
        onClick={onClick}
        className={gClasses.smallDisplay}
        style={{
          paddingBottom: "1rem",
        }}
      >
        <Image
          alt="SG logo"
          src={Logotipas_Oranzine}
          height={logoSize + 10}
          width={((logoSize + 10) / 785) * 2000}
          unoptimized
        />
      </Link>
      <Link href="/" onClick={onClick} className={gClasses.bigDisplay}>
        <Image
          alt="SG logo"
          src={HLogotipas_Oranzine}
          height={logoSize}
          width={(logoSize / 341) * 2000}
          unoptimized
        />
      </Link>
    </>
  );
}
