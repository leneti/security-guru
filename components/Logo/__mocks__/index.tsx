export function Logo(props: any) {
  return (
    <a data-testid="logo" {...props}>
      Mocked Logo
    </a>
  );
}
