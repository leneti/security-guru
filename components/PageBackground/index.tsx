export { default, type PageBGProps } from "./PageBackground";
