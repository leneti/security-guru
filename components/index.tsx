export { default as ContactInfo } from "./ContactInfo";
export { default as Footer } from "./Footer";
export { default as GetInTouch, type ContactForm } from "./GetInTouch";
export { default as Header } from "./Header";
export { default as Hero } from "./Hero";
export { default as Logo } from "./Logo";
export { default as PageTitle } from "./PageTitle";
export { default as PageBackground, type PageBGProps } from "./PageBackground";
export { default as SlideDownTitle } from "./SlideDownTitle";
