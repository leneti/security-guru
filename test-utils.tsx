import type { ReactElement, ReactNode } from "react";
import { MantineProvider } from "@mantine/core";
import { render, type RenderOptions } from "@testing-library/react";
import { resolver, theme } from "@site/constants";

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return (
    <MantineProvider theme={theme} cssVariablesResolver={resolver}>
      {children}
    </MantineProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
