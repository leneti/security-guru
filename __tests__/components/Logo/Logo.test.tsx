import { render } from "@site/test-utils";
import { Logo } from "@site/components/Logo";

describe("Logo", () => {
  it.each([
    { variant: "default", props: {} },
    { variant: "icon only", props: { iconOnly: true } },
    { variant: "vertical", props: { vertical: true } },
  ])("renders $variant", ({ props }) => {
    const { container } = render(<Logo {...props} />);
    expect(container).toMatchSnapshot();
  });
});
