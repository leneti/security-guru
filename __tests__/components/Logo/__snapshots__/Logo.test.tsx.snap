// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders default 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Security Guru logotipas"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/img.jpg"
      style="color: transparent; height: 3.125rem; width: 18.328445747800586rem;"
      width="293.2551319648094"
    />
  </a>
</div>
`;

exports[`Logo renders icon only 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Security Guru logotipas"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/img.jpg"
      style="color: transparent; height: 3.125rem; width: 3.1234382808595704rem;"
      width="49.97501249375313"
    />
  </a>
</div>
`;

exports[`Logo renders vertical 1`] = `
<div>
  <a
    href="/"
  >
    <img
      alt="Security Guru logotipas"
      data-nimg="1"
      decoding="async"
      height="50"
      loading="lazy"
      src="/img.jpg"
      style="color: transparent; height: 3.125rem; width: 7.961783439490445rem;"
      width="127.38853503184713"
    />
  </a>
</div>
`;
